packages/cli/src/fulltext/FileIndex.ts

```diff
@@ -49,32 +50,34 @@ export class FileIndex {
   }
 
   async indexDirectories(directories: string[]) {
-    for (const directory of directories) {
-      const gitState = await Git.state(directory);
-
-      let fileNames: string[];
-      if (gitState === GitState.Ok) {
-        fileNames = await listGitProjectFiles(directory);
-      } else {
-        fileNames = await listProjectFiles(directory);
-      }
-      this.#indexDirectory(directory, await filterFiles(directory, fileNames));
-    }
+    return Promise.all(
+      directories.map((directory) =>
```

Comment:
Running this synchronously does appear to be ~8% faster, so I'll switch it over. I'm kind of surprised by that actually.

Reply:
It may be faster @ the kernel level to iterate through directories in a file rather than jumping around between directories. (hitting the memory cache more?)

Reply:
Can you try this as a for loop and using async io inside the loop? If that’s as good as using synchronous read, then I’d prefer it.

Reply:
Serial calls to an asynchronous `indexFile` incurs a lot of overhead. These numbers don't include calls to `querySymbols`.

```
async indexDirectories -> Promise.all
 sync indexDirectory
 sync indexFile
-> 4779ms

async indexDirectories -> for / await
 sync indexDirectory
 sync indexFile
-> 5510ms

async indexDirectories -> for / await
async indexDirectory -> for / await
async indexFile
-> 18925ms

async indexDirectories -> Promise.all
async indexDirectory -> Promise.all
async indexFile
-> 5757ms
```

Reply:
The problem with option (2) is that the whole process (e.g. including other RPC calls) will be blocked until all directories have been processed. So I've added a proposal whereby:

* The file read occurs synchronously, but are yields to the event loop to avoid blocking the process.
* An exception that might occur within the directory indexing is caught and logged.